# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/css/css-contain/content-visibility/content-visibility-auto-selection-crash.html
# Commit: 4fca9cae3f42
# Full Hash: 4fca9cae3f42010acd2f985f54aa969d3e7fff5e
# Author: Martin Robinson <mrobinson@igalia.com>
# Date: 2023-01-04 04:29:41
# Description:
#   Bug 1807769 - Null-check frame selection in nsIFrame::HasSelectionInSubtree r=emilio
#   
#   Be more careful when accessing the frame selection, which might be
#   nullptr.
#   
# ==============================================================================

diff -r 0910d52020ec -r 4fca9cae3f42 testing/web-platform/tests/css/css-contain/content-visibility/content-visibility-auto-selection-crash.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/testing/web-platform/tests/css/css-contain/content-visibility/content-visibility-auto-selection-crash.html	Tue Jan 03 11:41:09 2023 +0000
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <link rel="author" title="Martin Robinson" href="mailto:mrobinson@igalia.com">
+    <link rel="help" href="https://drafts.csswg.org/css-contain/#content-visibility">
+    <meta name="assert" content="Changes to selection that result in no selection do not cause a crash.">
+    <style>
+        * {
+            content-visibility: auto;
+        }
+    </style>
+    <script>
+        document.addEventListener('DOMContentLoaded', () => {
+          document.documentElement.convertPointFromNode({}, marquee, {})
+          textarea.setSelectionRange(0, 0, 'forward')
+          input.setAttribute('type', 'number')
+          document.documentElement.getBoundingClientRect()
+        })
+    </script>
+</head>
+<body>
+    <marquee id='marquee'></marquee>
+    <textarea id='textarea'></textarea>
+    <input id='input'></input>
+</body>
+</html>
